forecasttools/idata_w_dates_to_df.py

```diff
@@ -7,80 +7,226 @@
+    idata: az.InferenceData,
+    group: str,
+    variable: str,
+    dimension: str,
+    start_date_iso: datetime | date,
+    time_step: timedelta,
 ) -> az.InferenceData:
     """
-    Modifies the provided idata object by assigning
-    date arrays to each group specified in group_dim_dict.
+    Adds time coordinates to a specified
+    variable within a group in an ArviZ
+    InferenceData object. This function
+    assigns a range of time coordinates
+    to a specified dimension in a variable
+    within a group in an InferenceData object.
 
     Parameters
     ----------
-    idata
-        The InferenceData object that contains
-        multiple groups (e.g., observed_data,
-        posterior_predictive).
-    group_dim_dict
-        A dictionary that maps InferenceData group
-        names (e.g., "observed_data", "posterior_predictive"
-        ) to dimension names (e.g., "obs_dim_0").
-    start_date_iso
-        The start date in ISO format (YYYY-MM-DD) from
-        which to begin the date range for the dimension.
+    idata : az.InferenceData
+        The InferenceData object containing
+        the group and variable to modify.
+    group : str
+        The name of the group within the
+        InferenceData object (e.g.,
+        "posterior_predictive").
+    variable : str
+        The name of the variable within the
+        specified group to assign time
+        coordinates to.
+    dimension : str
+        The dimension name to which time
+        coordinates should be assigned.
+    start_date_iso : date | datetime
+        The start date for the time
+        coordinates as a str in ISO format
+        (e.g., "2022-08-20") or as a
+        datetime object.
+    time_step : timedelta
+        The time interval between each
+        coordinate (e.g., `timedelta(days=1)`
+        for daily intervals).
+
 
     Returns
     -------
-    idata
-        The modified InferenceData object with
-        date coordinates assigned to each group.
+    az.InferenceData
+        The InferenceData object with updated
+        time coordinates for the specified
+        group, variable, and dimension.
     """
+    inputs = [
+        (idata, az.InferenceData, "idata"),
+        (group, str, "group"),
+        (variable, str, "variable"),
+        (dimension, str, "dimension"),
+        (time_step, timedelta, "time_step"),
+        (start_date_iso, (date, datetime), "start_date_iso"),
+    ]
+    for value, expected_type, param_name in inputs:
+        forecasttools.validate_input_type(
+            value=value, expected_type=expected_type, param_name=param_name
+        )
+    idata_group = forecasttools.validate_and_get_idata_group(
+        idata=idata, group=group
+    )
+    variable_data = forecasttools.validate_and_get_idata_group_var(
+        idata_group=idata_group, group=group, variable=variable
+    )
+    forecasttools.validate_idata_group_var_dim(
+        variable_data=variable_data, dimension=dimension
+    )
```

Comment:
Beyond the scope of this PR, but once this lands there should be a cleanup PR to remove these validation helpers from the top-level namespace. Even if they're going to be user-exposed, that's too prominent I think.